google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ModelItem.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.resources;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+
+/**
+ * Represents an existing or non-existing GCP project
+ */
+public class ModelItem extends DefaultMutableTreeNode {
```

Comment:
Why the need to add this in between?

Also the name of it doesn't really indicate that its a GCP related object.

Reply:
I added the ModelItem, so that we can tell when the project field has no value (null), an invalid project(InvalidProjectModelItem) and a valid project (ResourceProjectModelItem).

I agree with the name comment. Any suggestions? I am not a good name-generator :)